app/models/spree/gateway/stripe_connect.rb

```diff
@@ -0,0 +1,133 @@
+        'stripe'
+      end
+
+      def provider_class
+        ActiveMerchant::Billing::StripeGateway
+      end
+
+      def payment_profiles_supported?
+        true
+      end
+
+      def stripe_account_id
+        StripeAccount.find_by_enterprise_id(preferred_enterprise_id).andand.stripe_user_id
+      end
+
+      def purchase(money, creditcard, gateway_options)
+        provider.purchase(*options_for_purchase_or_auth(money, creditcard, gateway_options))
+      end
+
+
+      def create_profile(payment)
+        return unless payment.source.gateway_customer_profile_id.nil?
+        options = {
+          email: payment.order.email,
+          login: Stripe.api_key,
+        }.merge! address_for(payment)
+
+        creditcard = card_to_store(payment.source)
+
+        response = provider.store(creditcard, options)
+        if response.success?
+          payment.source.update_attributes!( cc_type: payment.source.cc_type, # side-effect of update_source!
+                                             gateway_customer_profile_id: response.params['id'],
+                                             gateway_payment_profile_id: response.params['default_source'] || response.params['default_card'])
+        else
+          payment.send(:gateway_error, response.message)
+        end
+      end
+
+      private
+
+      # In this gateway, what we call 'secret_key' is the 'login'
+      def options
+        options = super
+        options.merge(:login => Stripe.api_key)
+      end
+
+      def options_for_purchase_or_auth(money, creditcard, gateway_options)
+        options = {}
+        options[:description] = "Spree Order ID: #{gateway_options[:order_id]}"
+        options[:currency] = gateway_options[:currency]
+        options[:stripe_account] = stripe_account_id
+
+        creditcard = token_from_card_profile_ids(creditcard)
+
+        [money, creditcard, options]
+      end
+
+      def address_for(payment)
+        {}.tap do |options|
+          if address = payment.order.bill_address
+            options[:address] = {
+              address1: address.address1,
+              address2: address.address2,
+              city: address.city,
+              zip: address.zipcode
+            }
+
+            if country = address.country
+              options[:address][:country] = country.name
+            end
+
+            if state = address.state
+              options[:address].merge!(state: state.name)
+            end
+          end
+        end
+      end
+
+      def update_source!(source)
+        source.cc_type = CARD_TYPE_MAPPING[source.cc_type] if CARD_TYPE_MAPPING.include?(source.cc_type)
+        source
+      end
+
+      def card_to_store(source)
+        source = update_source!(source)
+        if source.number.blank? && source.gateway_payment_profile_id.present?
+          # StripeJS Token
+          source.gateway_payment_profile_id
+        else
+          # Spree::CreditCard object
+          source
+        end
+      end
```

Comment:
Probably in the future we'll need some kind of `CreditCard` object that could incapsulate all this logic.